src/backersManager/rewardDistributionStarted.ts

```diff
@@ -1,21 +1,23 @@
 import { RewardDistributionStarted as RewardDistributionStartedEvent } from "../../generated/BackersManagerRootstockCollective/BackersManagerRootstockCollective";
-import { ContractConfig } from "../../generated/schema";
 import { RewardDistributorRootstockCollective as RewardDistributorRootstockCollectiveContract } from "../../generated/RewardDistributorRootstockCollective/RewardDistributorRootstockCollective";
-import { Address } from "@graphprotocol/graph-ts";
-import { loadOrCreateCycle, CONTRACT_CONFIG_ID, logEntityNotFound, updateBlockInfo } from "../utils";
+import { BackersManagerRootstockCollective as BackersManagerRootstockCollectiveContract } from "../../generated/BackersManagerRootstockCollective/BackersManagerRootstockCollective";
+import { Address, Bytes } from "@graphprotocol/graph-ts";
+import { loadOrCreateCycle, updateBlockInfo, loadOrCreateContractConfig } from "../utils";
 
 export function handleRewardDistributionStarted(
   event: RewardDistributionStartedEvent
 ): void {
-  const cycle = loadOrCreateCycle(event.address);
+  const backersManagerContract = BackersManagerRootstockCollectiveContract.bind(
+    event.address
+  );
+  const cycleStart = backersManagerContract.cycleStart(event.block.timestamp);
+  const cycle = loadOrCreateCycle(changetype<Bytes>(Bytes.fromBigInt(cycleStart)));
```

Comment:
should we identify it as a progressive number?

Reply:
I don't understand the question :S